server/models/dataset_helpers/image_metadata.py

```diff
@@ -260,8 +261,12 @@ class PersonalHxMmFieldParser(HxMmFieldParser):
 
 
 class ClinicalSizeFieldParser(FieldParser):
+    """
+    Parse clinical size field. Expects units to be specified (um, mm, or cm).
+    """
     name = 'clin_size_long_diam_mm'
     allowedFields = {'clin_size_long_diam_mm'}
+    _formatRegex = re.compile(r'(.+)\s?(um|mm|cm)$')
```

Comment:
There's not really a point to this single `\s?` as-is, since other whitespace characters can be matched by `.+` and (multiple) whitespaces can be passed into the [`float` function](https://docs.python.org/2/library/functions.html#float). Either more strict number/whitespace matching should be done in this regex, or it should all be done by `_coerceFloat`. IMO, we should do the latter, since that's more general/shared code.